org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/generators/xml2dtd/XML2DTDGeneratorTest.java

```diff
@@ -241,4 +260,59 @@ public void oneZero2() {
 				"<!ELEMENT c EMPTY>";
 		assertGrammarGenerator(xml, new DTDGeneratorSettings(), dtd);
 	}
+
+	@Test
+	public void optionalAttribute() {
+		String xml = "<root>\r\n" + //
+				"	<item attr1=\"A\" attr2=\"B\"/>\r\n" + //
+				"	<item attr1=\"A\" />\r\n" + //
+				"</root>";
+		String dtd = "<!ELEMENT root (item+)>" + lineSeparator() + //
+				"<!ELEMENT item EMPTY>" + lineSeparator() + //
+				"<!ATTLIST item attr1 NMTOKEN #REQUIRED>" + lineSeparator() + //
+				"<!ATTLIST item attr2 NMTOKEN #IMPLIED>";
+		assertGrammarGenerator(xml, new DTDGeneratorSettings(), dtd);
+	}
+
+	@Test
+	public void attrIDsAndFixed() {
+		String xml = "<root>\r\n" + //
+				"	<item attr1=\"id1\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id2\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id3\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id4\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id5\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id6\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id7\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id8\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id9\" attr2=\"A\" />\r\n" + //
+				"	<item attr1=\"id10\" attr2=\"A\" />\r\n" + //
+				"</root>";
+		String dtd = "<!ELEMENT root (item+)>" + lineSeparator() + //
+				"<!ELEMENT item EMPTY>" + lineSeparator() + //
+				"<!ATTLIST item attr1 ID #REQUIRED>" + lineSeparator() + //
+				"<!ATTLIST item attr2 NMTOKEN #FIXED A>";
+		assertGrammarGenerator(xml, new DTDGeneratorSettings(), dtd);

```

Comment:
For this use case, have a strange error message in my XML after generating the DTD:

![image](https://user-images.githubusercontent.com/20326645/85612725-20e3c800-b627-11ea-92b9-64eab10e38aa.png)

Reply:
fixed